backend/api/EventHandlers/MissionScheduler.cs

```diff
@@ -32,9 +32,9 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
         {
             while (!stoppingToken.IsCancellationRequested)
             {
-                foreach (var upcomingMission in UpcomingMissions)
+                foreach (var queuedMission in MissionQueue)
                 {
-                    var freshMission = MissionService.ReadById(upcomingMission.Id).Result;
+                    var freshMission = MissionService.ReadById(queuedMission.Id).Result;
```

Comment:
Out of scope of this PR - But I would rename this variable to a more descriptive name.